# Upstream patch to fix some more encoding errors with libavformat v56.
# URL: https://github.com/mltframework/mlt/commit/e8c0fb4531cb4de142cebd7a39ee7863b76e7e03

diff -Naur mlt-0.9.2.orig/src/modules/avformat/consumer_avformat.c mlt-0.9.2/src/modules/avformat/consumer_avformat.c
--- mlt-0.9.2.orig/src/modules/avformat/consumer_avformat.c	2014-09-24 18:52:22.258010988 +0200
+++ mlt-0.9.2/src/modules/avformat/consumer_avformat.c	2014-09-24 18:53:46.878008587 +0200
@@ -1801,7 +1801,11 @@
 
 						if ( i == 0 )
 						{
+#if LIBAVFORMAT_VERSION_MAJOR >= 56
+							audio_pts = (double) sample_count[0] * av_q2d( stream->codec->time_base );
+#else
 							audio_pts = (double) sample_count[0] * av_q2d( stream->time_base );
+#endif
 						}
 					}
 				}
@@ -1991,7 +1995,11 @@
 						}
  					}
 					frame_count++;
+#if LIBAVFORMAT_VERSION_MAJOR >= 56
+					video_pts = (double) frame_count * av_q2d( video_st->codec->time_base );
+#else
 					video_pts = (double) frame_count * av_q2d( video_st->time_base );
+#endif
 					if ( ret )
 					{
 						mlt_log_fatal( MLT_CONSUMER_SERVICE( consumer ), "error writing video frame\n" );
